libcaf_core/src/actor_system_config.cpp

```diff
@@ -516,6 +521,7 @@ error actor_system_config::extract_config_file_path(string_list& args) {
   auto evalue = ptr->parse(path);
   if (!evalue)
     return std::move(evalue.error());
+  put(content, "config-file", *evalue);
```

Comment:
How does this relate to the addresses issue? Parse already failed here if the config file is not available.

Reply:
`extract_config_file_path` gets called before parsing any other CLI option. The work flow is basically:
- check if there's a `--config-file=` in the CLI arguments
- if yes: parse that argument (but not any other)
- read the config file
- read the CLI arguments

By putting `config-file` into our `content` map, we have access to the information whether the user has requested to read that file or if we simply try opening the default.

Reply:
Ok.